### PR TITLE
28573 wrong thumbnail

### LINKED ISSUE #28573: GraphQL - ConfigurableCartItem added to Cart returns wrong thumbnail image
If you add a *ConfigurableCartItem* to the Cart and then Query the Cart, this returns the Thumbnail of the Parent Product instead of the Variant that was selected.
Architecture proposal magento/architecture#380

**Query**
```java
query CartQuery($cartId: String!) {
  cart(cart_id: $cartId) {
    ...Cart
  }
}

fragment Cart on Cart {
  items {
    id
    quantity
    product {
      id

      thumbnail {
        label
        url
      }
    }
    ... on ConfigurableCartItem {
      configurable_options {
        id
        option_label
        value_label
      }
    }
  }
}
{code}
**Variables**
{code:java}
{
  "cartId": "w74YbL1jbhlrdS95vWQM1mktppa9M3wI"
}
{code}
**Result**
{code:java}
{
  "data": {
    "cart": {
      "items": <
        {
          "id": "241",
          "quantity": 1,
          "product": {
            "id": 1410,
            "thumbnail": {
              "label": "Olivia 1/4 Zip Light Jacket",
              "url": "https://carlos-233-3h6p5qa-a6terwtbk67os.demo.magentosite.cloud/media/catalog/product/cache/ca63cacbf9a8041003a3fd3abbbfe421/w/j/wj12-blue_main.jpg" Should be the Thumbnail for Purple Variant
            }
          },
          "configurable_options": [
            {
              "id": 93,
              "option_label": "Color",
              "value_label": "Purple"
            },
            {
              "id": 162,
              "option_label": "Size",
              "value_label": "XS"
            }
          ]
        }
      >
    }
  }
}
```

### PR DESCRIPTION
**Pull request by sasha19957099**
### Description (*)
improved performance and test coverage

### Related Pull Requests
Based on https://github.com/magento/magento2/pull/30817

### Fixed Issues (if relevant)
1. Fixes https://github.com/magento/magento2/issues/28573

### Manual testing scenarios (*)
1. Configure checkout/cart/configurable_product_image as 'itself'
2. Run the following mutation:
mutation {
  addConfigurableProductsToCart(
    input:{
      cart_id:"{$maskedQuoteId}"
      cart_items:{
        parent_sku: "{$parentSku}"
        data:{
          sku:"{$sku}"
          quantity:{$quantity}
        }
      }
    }
  ) {
    cart {
      items {
        id
        quantity
        product {
          sku
        }
        ... on ConfigurableCartItem {
          configured_variant {
            sku
            thumbnail {
              label
              url
            }
          }
        }
      }
    }
  }
}

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by sasha19957099** [comment]
@magento run all tests

**Comment by sasha19957099** [comment]
@magento run Functional Tests B2B
@magento run Functional Tests EE

**Comment by sasha19957099** [comment]
@magento run Functional Tests EE

File: app/code/Magento/ConfigurableProductGraphQl/etc/schema.graphqls [hunk 1]
```diff
@@ -64,6 +64,7 @@ input ConfigurableProductCartItemInput {
 type ConfigurableCartItem implements CartItemInterface {
     customizable_options: [SelectedCustomizableOption] @resolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CustomizableOptions")
     configurable_options: [SelectedConfigurableOption!]! @resolver(class: "Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\ConfigurableCartItemOptions")
+    configured_variant: ProductInterface @doc(description: "Product details of the cart item") @resolver(class: "\\Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\ProductResolver")
```
Comment (prabhuram93) [inline app/code/Magento/ConfigurableProductGraphQl/etc/schema.graphqls]: Hey @sasha19957099 , any reasons for deviating from the schema here ?

https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/cart/Cart.graphqls

Comment (prabhuram93) [reply app/code/Magento/ConfigurableProductGraphQl/etc/schema.graphqls]: Hey @cpartica. I have 2 questions on this
1.  Do we need to expose store config to expose this configuration ```checkout/cart/configurable_product_image either parent or itself```. 
2. configured_variant should return ProductInterface or SimpleProduct! as per the schema?

Comment (prabhuram93) [reply app/code/Magento/ConfigurableProductGraphQl/etc/schema.graphqls]: Resolution: 
1. in `configured_variant: ProductInterface` ProductInterface should be mandatory "!"
2. `configurable_thumbnail_source` is already exposed in store config.

File: app/code/Magento/ConfigurableProductGraphQl/etc/schema.graphqls [hunk 2]
```diff
@@ -64,6 +64,7 @@ input ConfigurableProductCartItemInput {
 type ConfigurableCartItem implements CartItemInterface {
     customizable_options: [SelectedCustomizableOption] @resolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CustomizableOptions")
     configurable_options: [SelectedConfigurableOption!]! @resolver(class: "Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\ConfigurableCartItemOptions")
+    configured_variant: ProductInterface! @doc(description: "Product details of the cart item") @resolver(class: "\\Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\ProductResolver")
```
Comment (cpartica) [inline app/code/Magento/ConfigurableProductGraphQl/etc/schema.graphqls:67]: @keharper can you suggest a better doc here?

**Comment by sasha19957099** [comment]
@magento run all tests

**Comment by sasha19957099** [comment]
@magento run Functional Tests B2B

**Comment by sasha19957099** [comment]
@magento run Functional Tests CE

**Comment by sasha19957099** [comment]
@magento run Functional Tests EE

**Comment by sasha19957099** [comment]
@magento run WebAPI Tests

**Comment by sasha19957099** [comment]
@magento run Functional Tests B2B

**Comment by sasha19957099** [comment]
@magento run Functional Tests B2B

**Review by prabhuram93 — APPROVED** [review]

**Comment by prabhuram93** [comment]
This pull request is being processed internally. Please do not add any more commits on this PR.

**Pull request merged**